src/app/components/shared/datetime/datetime/datetime.component.spec.ts

```diff
@@ -0,0 +1,230 @@
+import { Spectator, createComponentFactory } from "@ngneat/spectator";
+import { SharedModule } from "@shared/shared.module";
+import { MockBawApiModule } from "@baw-api/baw-apiMock.module";
+import { DateTime, Settings } from "luxon";
+import { assertTooltip } from "@test/helpers/html";
+import { DatetimeComponent } from "./datetime.component";
+
+// I have created this interface for TypeScript LSP typing and auto completion
+// it should not be used outside of this file
+interface TestCase {
+  name: string;
+  value: Date | DateTime;
+  expectedText: string;
+  expectedTooltip: string;
+  expectedDateTimeAttribute: string;
+  dateOnly?: boolean;
+  timeOnly?: boolean;
+}
+
+// by setting the defaultZone to utc when we create DateTime objects from ISO strings
+// it doesn't implicitly localize the DateTime object to another timezone
+// this makes the object produced by the iso string predictable
+// eg. an iso string "2020-10-10T12:12:12Z" will always produce the object
+//     { year: 2020, month: 10, day: 10, hour: 12, minute: 12, second: 12 }
+// we re-mock the test runners timezone to a more complex one when the tests are run
+// however, during test setup it is easier to read the DateTime objects if we create
+// them as if the test runner is in utc+00:00
+Settings.defaultZone = "utc";
+
+// in the following test cases, all the dateTime objects are created in UTC+00:00
+// therefore, because we are mocking the test runners timezone to be Australia/Perth (UTC+08:00)
+// we should see that the date time text, tooltips and machine readable content is localized
+// to UTC+08:00. eg. We should see the hours increase by 8
+const testCases: TestCase[] = [
```

Comment:
greatly simplify the names of these test cases.

Data driven test cases don't usually have names. They're a grouping only with useful diagnostic information. If they're long sentences the reader had to parse them to understand the differences.